lab5/utils/CollectionManager.java

```diff
@@ -0,0 +1,390 @@
+package org.example.utils;
+
+//import jakarta.xml.bind.annotation.XmlAccessType;
+//import jakarta.xml.bind.annotation.XmlAccessorType;
+//import jakarta.xml.bind.annotation.XmlElement;
+//import jakarta.xml.bind.annotation.XmlRootElement;
+import org.example.collections.*;
+
+//import javax.xml.namespace.QName;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+
+@XmlRootElement(name = "list")
+@XmlAccessorType(XmlAccessType.FIELD)
+public class CollectionManager {
+    @XmlElement(name = "city")
+    private final ArrayDeque<City> cities = new ArrayDeque<>();
+
+    public ArrayDeque<City> getCities() {
+        return cities;
+    }
+
+    public void setCities(ArrayDeque<City> loadedCollection) {
+        if(loadedCollection != null) {
+            cities.clear();
+            cities.addAll(loadedCollection);
+        } else {
+            System.out.println("Warning: Attempted to load null collection");
+        }
+    }
+
+    private void sortCollection(){
+        List<City> cityList = new ArrayList<>(cities);
+        cityList.sort(Comparator.comparingInt(City::getId));
```

Comment:
This is incorrect

Your objects must either have a Comparator for them or implement Comparable